app/controllers/miq_ae_class_controller.rb

```diff
@@ -172,6 +172,9 @@ def get_node_info(node)
       get_method_node_info(id)
     when "aen"
       @record = MiqAeNamespace.find_by_id(from_cid(id[1]))
+      # need to set record as Domain record if it's a domain, editable_domains, enabled_domains,
+      # visible domains methods returns list of Domains, need this for toolbars to hide/disable correct records.
+      @record = MiqAeDomain.find_by_id(from_cid(id[1])) if @record.domain?
```

Comment:
Should this logic live in the model?  I have no idea what this means, it just seems strange to live here.